Jellyfin.Server.Implementations/Users/UserManager.cs

```diff
@@ -0,0 +1,818 @@
+        /// Initializes a new instance of the <see cref="UserManager"/> class.
+        /// </summary>
+        /// <param name="dbProvider">The database provider.</param>
+        /// <param name="cryptoProvider">The cryptography provider.</param>
+        /// <param name="networkManager">The network manager.</param>
+        /// <param name="appHost">The application host.</param>
+        /// <param name="imageProcessor">The image processor.</param>
+        /// <param name="logger">The logger.</param>
+        public UserManager(
+            JellyfinDbProvider dbProvider,
+            ICryptoProvider cryptoProvider,
+            INetworkManager networkManager,
+            IApplicationHost appHost,
+            IImageProcessor imageProcessor,
+            ILogger<IUserManager> logger)
+        {
+            _dbProvider = dbProvider;
+            _cryptoProvider = cryptoProvider;
+            _networkManager = networkManager;
+            _appHost = appHost;
+            _imageProcessor = imageProcessor;
+            _logger = logger;
+        }
+
+        /// <inheritdoc/>
+        public event EventHandler<GenericEventArgs<User>> OnUserPasswordChanged;
+
+        /// <inheritdoc/>
+        public event EventHandler<GenericEventArgs<User>> OnUserUpdated;
+
+        /// <inheritdoc/>
+        public event EventHandler<GenericEventArgs<User>> OnUserCreated;
+
+        /// <inheritdoc/>
+        public event EventHandler<GenericEventArgs<User>> OnUserDeleted;
+
+        /// <inheritdoc/>
+        public event EventHandler<GenericEventArgs<User>> OnUserLockedOut;
+
+        /// <inheritdoc/>
+        public IEnumerable<User> Users
+        {
+            get
+            {
+                var dbContext = _dbProvider.CreateContext();
+                return dbContext.Users;
+            }
+        }
+
+        /// <inheritdoc/>
+        public IEnumerable<Guid> UsersIds
+        {
+            get
+            {
+                var dbContext = _dbProvider.CreateContext();
+                return dbContext.Users.Select(u => u.Id);
+            }
+        }
+
+        /// <inheritdoc/>
+        public User GetUserById(Guid id)
+        {
+            if (id == Guid.Empty)
+            {
+                throw new ArgumentException("Guid can't be empty", nameof(id));
+            }
+
+            var dbContext = _dbProvider.CreateContext();
+
+            return dbContext.Users.Find(id);
+        }
+
+        /// <inheritdoc/>
+        public User GetUserByName(string name)
+        {
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                throw new ArgumentException("Invalid username", nameof(name));
+            }
+
+            var dbContext = _dbProvider.CreateContext();
+
+            // This can't use an overload with StringComparer because that would cause the query to
+            // have to be evaluated client-side.
+            return dbContext.Users.FirstOrDefault(u => string.Equals(u.Username, name));
```

Comment:
Does the username field use `NOCASE` for Sqlite?

Reply:
I never specified anything, so probably not

Reply:
This should be added. I agree that we shouldn't let users of the same name but different case exist.

Reply:
For that to be enforced, changes made at user retrieval would be irrelevant, we would have to enforce that at user creation time

Reply:
Right, but nocase storage would allow a comparison to the provided username string and usernames in the database of any casing. Example would be `SpArKy`. That's a hassle to check with just SQL.

If you don't use SQL for that, it will require to to retrieve the list of all users, to lowercase it all, to lowercase the provided user string, etc. If you use SQL for this you just say `select * from Users where username like "username"` and it'll do a comparison that ignores casing. If you get a result back, you can't proceed as it already exists. Lot simpler.